### ISSUE TITLE
LwF plugin fails in combination with IncrementalClassifier

### ISSUE THREAD

**Issue by AlbinSou** [ContinualAI/avalanche#746]
🐛 **Describe the bug**
LwFPlugin does not work when combined with IncrementalClassifier.

The error raised is the following (raised when starting to train on second experience):
in lwf.py - _distillation_loss -> line 45
The size of tensor a (6) must match the size of tensor b (10) at non-singleton dimension 1

I believe this is due to the fact that the size of the previous model output and the current model output are different.

I initially thought that this occurred because the option class_ids_from_zero_from_first_exp was not activated, which makes the IncrementalClassifier head as big as the maximum class value (I think it should be documented that IncrementalClassifier should be used in conjuction with this option). But in the end, even when turning this option on, the bug was still appearing but this time with a logical increment between the two output sizes (prev_out.size(0) = out.size(0) - n_classes_in_exp).

🐜 **To Reproduce**
In the example examples/lwf_mnist.py, change the model classifier to an IncrementalClassifier

```{python}
model = SimpleMLP(hidden_size=args.hs)
model.classifier = IncrementalClassifier(model.classifier.in_features)
```

🐝 **Expected behavior**
I think it should be possible to combine this plugin with this classifier type, either by masking the new logits when computing the loss, or by continuing to adapt the head of the old model. At least an error should be raised if the combination is not deemed possible.
